lib/src/layout/window.dart

```diff
@@ -95,7 +95,12 @@ class _MacosWindowState extends State<MacosWindow> {
     late Color sidebarBackgroundColor;
     Color dividerColor = theme.dividerColor;
 
-    if (!theme.brightness.isDark) {
+    // Only show blurry, transparent sidebar when platform brightness and app
+    // brightness are the same, otherwise it looks awful
+    if (MediaQuery.of(context).platformBrightness.isDark ==
+        theme.brightness.isDark) {
+      sidebarBackgroundColor = Colors.transparent;
+    } else if (!theme.brightness.isDark) {
       sidebarBackgroundColor = widget.sidebar?.decoration?.color ??
           CupertinoColors.systemGrey6.color;
     } else {
```

Comment:
2 things:

- This doesn't look right on web and Windows (I didn't test Linux). Screenshots below.
- I would expect any color override from the parent to take precedence, but right now it will be overridden by the transparency effect.

I would suggest the following changes to fix both issues:

```suggestion
    final isMac = !kIsWeb && defaultTargetPlatform == TargetPlatform.macOS;
    
    // Respect the sidebar color override from parent if one is given
    if (widget.sidebar?.decoration?.color != null) {
      sidebarBackgroundColor = widget.sidebar!.decoration!.color!;
    } else if (isMac &&
        MediaQuery.of(context).platformBrightness.isDark ==
            theme.brightness.isDark) {
      // Only show blurry, transparent sidebar when platform brightness and app
      // brightness are the same, otherwise it looks awful. Also only make the
      // sidebar transparent on native Mac, or it will just be flat black or
      // white.
      sidebarBackgroundColor = Colors.transparent;
    } else if (!theme.brightness.isDark) {
      sidebarBackgroundColor = widget.sidebar?.decoration?.color ??
          CupertinoColors.systemGrey6.color;
    } else {
      sidebarBackgroundColor = widget.sidebar?.decoration?.color ??
          CupertinoColors.tertiarySystemBackground.darkColor;
}
```

This change is meant to replace line 97 - 109 rather than 97 - 106, but Github won't let me apply the suggestion that far outside the diff 😒.

Web:

![image](https://user-images.githubusercontent.com/1316184/150372855-2a2dc59b-e507-46b3-9380-83d660bcd318.png)

Windows:

![image](https://user-images.githubusercontent.com/1316184/150372660-dd9cbe71-d424-47bc-94b0-4b60590fbf8b.png)